### DISCUSSION TITLE
Isn't the team too small ?

### DISCUSSION THREAD

**Discussion by MhAllan** [dotnet/maui#14354] [General]
Looking at the Microsoft team working on Maui and Xamarin I think it looks like one small team working on everything. I guess we have 2k+ issues and with these platforms keep evolving, it is unlikely this number will go down. I doubt we need to have at least 3 big teams as in one team for each platform Android, IOS, and Windows.

**Comment by ivmirx** [comment]
I agree, but I'm afraid MS management got trapped in a vicious circle.

The tech world likes chasing hypes – see a new trendy JS framework every few years. So, when something like Flutter releases, it's automatically better because it's a New Technology That Solves Everything. No amount of rebranding Xamarin or Forms will produce the same kind of effect even inside the .NET world. And that's even before the fact that Flutter indeed receives more tech and marketing resources from Google from the very beginning.

So why invest more resources into something that can't escape its niche and already has a bad reputation in the tech world at large? I remember people hating Xamarin and confusing it with Forms back in 2016, and back then there was no decent cross-platform technology whatsoever. Now they will just sneer and start pushing Flutter (despite it being on par with React Native in popularity, but RN is also old).

I believe that mobile .NET is doomed to remain an afterthought for niche enterprise goals, Unity, and whatnot. But I'd be happy to be wrong about that because the whole ecosystem is amazing and it not becoming a cross-platform standard feels like a huge missed opportunity.

**Reply by Xyncgas** [reply]
The people working inside programming gotta understand it's a program that needs to work not a fashion show that needs to be young and beautiful

Flutter is nice but I can use math to draw layout so how about making sure MAUI actually works instead of having 2k bugs and I can do everything myself

**Comment by MhAllan** [comment]
@ivmirx Thanks for your reply. I really think C# and Xaml are loved techs. and I believe Azure is ways better than any other clouds especially when it comes to productivity, integration, and simplicity. Microsoft techs in general are better than others, more clean and robust. Maui enables Azure, that is the key behind Maui and Xamarin, but for some reason they lost this goal! and everytime I want to adopt azure I find mobiles blocking me and so Xamarin/Maui are not helping me!

**Reply by Xyncgas** [reply]
Let me tell you a thing about cloud :
If you can cost nothing for computational resource, that's better
Apparently you can do that by computing everything in mobile, storing data in emulated file system in object storage that the user can pay for themselves, so you don't need to pay for anything for infrastructures and the users is happy they can choose their providers for hosting your app

Imagine in the future instead of renting out servers, clouds will rent out hosting services for your instagram account facebook account so all you have to do is make your own applications and people can find marketplace to host the app themselves, while the hosting services can use a trustless protocol standardized for developers, if you don't like letting other people help you host the applications that you use you can host the app yourself by renting a server and installing the environment as easy as we are installing docker today

Or store everything locally, exchanging data via peered transfer
